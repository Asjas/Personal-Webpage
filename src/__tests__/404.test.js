import React from 'react';
import { render } from 'react-testing-library';
import Fourohfour from '../pages/404';

describe('Fourohfour', () => {
  it('renders correctly', () => {
    const linkText = 'Go Home';
    const headingText = 'PAGE NOT FOUND';
    const location = {
      pathname: '/404',
    };

    const { container, getByTestId } = render(<Fourohfour location={location} />);
    expect(getByTestId('404-heading')).toHaveTextContent(headingText);
    expect(getByTestId('404-link')).toHaveTextContent(linkText);
    expect(container).toMatchSnapshot();
  });
});
