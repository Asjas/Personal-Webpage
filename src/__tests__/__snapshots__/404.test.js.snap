// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fourohfour renders correctly 1`] = `
.c0 {
  display: grid;
  grid-template-columns: 400px 1fr;
  grid-gap: 20px;
  grid-template-areas: 'aside main';
}

.c1 {
  width: 400px;
  position: fixed;
  grid-area: aside;
}

.c7 {
  grid-area: main;
}

.c2 {
  background: #222;
  border-top: 4px solid #FFD700;
  border-bottom: 4px solid #FFD700;
  height: 100vh;
  display: grid;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  color: #EDEDED;
  font-size: 2.2rem;
  line-height: 2.2rem;
  border-top: 2px solid rgba(255,255,255,0.04);
  border-bottom: 2px solid rgba(255,255,255,0.04);
  cursor: pointer;
  -webkit-letter-spacing: 0.03rem;
  -moz-letter-spacing: 0.03rem;
  -ms-letter-spacing: 0.03rem;
  letter-spacing: 0.03rem;
  padding: 20px 0;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
}

.c4:hover {
  color: #FFD700;
}

.c4:focus {
  border: 2px solid #FFD700;
  color: #FFD700;
  outline: none;
}

.c5 {
  display: grid;
  grid-template-columns: repeat(5,1fr);
  margin: 0 0 0 10px;
  max-width: 400px;
  -webkit-align-self: end;
  -ms-flex-item-align: end;
  align-self: end;
}

.c6 {
  cursor: pointer;
}

.c6:focus {
  outline: none;
}

.c6 svg {
  fill: #EDEDED;
  height: 5.5rem;
  width: 5.5rem;
}

.c6 svg:hover {
  fill: #FFD700;
}

.c6:focus svg {
  fill: #FFD700;
}

.c6 span {
  position: absolute;
  left: -9999px;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 65vh;
}

.c9 {
  font-size: 6.0rem;
  color: #222;
}

.c10 {
  font-size: 3.2rem;
  color: #222;
}

.c11 {
  font-size: 1.7rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #D2006B;
  padding: 3px;
}

.c11:hover {
  color: lightcoral;
}

.c11:focus {
  outline: 1px solid #D2006B;
}

@media (max-width:90em) {
  .c0 {
    grid-template-columns: 320px 1fr;
  }
}

@media (max-width:64em) {
  .c0 {
    grid-template-columns: 1fr;
    grid-template-areas: 'main';
  }
}

@media (max-width:90em) {
  .c1 {
    max-width: 320px;
  }
}

@media (max-width:64em) {
  .c1 {
    display: none;
  }
}

@media (max-width:90em) {
  .c5 {
    max-width: 320px;
  }
}

@media (max-width:90em) {
  .c6 svg {
    height: 5rem;
    width: 5rem;
  }
}

<div>
  <div
    class="c0"
  >
    <section
      class="c1"
    >
      <nav
        class="c2"
      >
        <section
          class="c3"
        >
          <a
            class="c4"
            href="/"
          >
            Home
          </a>
          <a
            class="c4"
            href="/portfolio/"
          >
            Portfolio
          </a>
          <a
            class="c4"
            href="/services/"
          >
            Services
          </a>
          <a
            class="c4"
            href="/contact/"
          >
            Contact
          </a>
        </section>
        <footer
          class="c5"
        >
          <a
            class="c6"
            color="#EDEDED"
            href="https://twitter.com/asjasroos"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              aria-labelledby="twitter-icon"
              viewBox="0 0 30 30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M23.95 4.57a10 10 0 0 1-2.82.77 4.96 4.96 0 0 0 2.16-2.72c-.95.56-2 .96-3.12 1.19a4.92 4.92 0 0 0-8.38 4.48A13.93 13.93 0 0 1 1.63 3.16a4.92 4.92 0 0 0 1.52 6.57 4.9 4.9 0 0 1-2.23-.61v.06c0 2.38 1.7 4.37 3.95 4.82a5 5 0 0 1-2.21.09 4.94 4.94 0 0 0 4.6 3.42A9.87 9.87 0 0 1 0 19.54a14 14 0 0 0 7.56 2.21c9.05 0 14-7.5 14-13.98 0-.21 0-.42-.02-.63A9.94 9.94 0 0 0 24 4.59l-.05-.02z"
              />
            </svg>
            <span>
              Link to twitter profile
            </span>
          </a>
          <a
            class="c6"
            color="#EDEDED"
            href="https://github.com/asjas"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              aria-labelledby="github-icon"
              viewBox="0 0 30 30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 .3a12 12 0 0 0-3.8 23.38c.6.12.83-.26.83-.57L9 21.07c-3.34.72-4.04-1.61-4.04-1.61-.55-1.39-1.34-1.76-1.34-1.76-1.08-.74.09-.73.09-.73 1.2.09 1.83 1.24 1.83 1.24 1.07 1.83 2.81 1.3 3.5 1 .1-.78.42-1.31.76-1.61-2.67-.3-5.47-1.33-5.47-5.93 0-1.31.47-2.38 1.24-3.22-.14-.3-.54-1.52.1-3.18 0 0 1-.32 3.3 1.23a11.5 11.5 0 0 1 6 0c2.28-1.55 3.29-1.23 3.29-1.23.64 1.66.24 2.88.12 3.18a4.65 4.65 0 0 1 1.23 3.22c0 4.61-2.8 5.63-5.48 5.92.42.36.81 1.1.81 2.22l-.01 3.29c0 .31.2.69.82.57A12 12 0 0 0 12 .3"
              />
            </svg>
            <span>
              Link to github profile
            </span>
          </a>
          <a
            class="c6"
            color="#EDEDED"
            href="https://medium.com/@asjas"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              aria-labelledby="medium-icon"
              viewBox="0 0 30 30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M2.85 6.36a.93.93 0 0 0-.3-.78L.3 2.88v-.4h6.96l5.38 11.79 4.73-11.8H24v.4l-1.92 1.84a.56.56 0 0 0-.21.54v13.5c-.03.2.05.41.21.54l1.87 1.84v.4h-9.4v-.4l1.93-1.89c.19-.19.19-.24.19-.53V7.78l-5.39 13.7h-.73L4.28 7.78v9.18c-.05.38.07.77.34 1.05l2.52 3.06v.4H0v-.4l2.52-3.06c.27-.28.39-.67.33-1.05V6.36z"
              />
            </svg>
            <span>
              Link to medium profile
            </span>
          </a>
          <a
            class="c6"
            color="#EDEDED"
            href="https://www.linkedin.com/in/asjasroos/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              aria-labelledby="linkedin-icon"
              viewBox="0 0 30 30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20.45 20.45h-3.56v-5.57c0-1.32-.02-3.03-1.85-3.03-1.85 0-2.13 1.44-2.13 2.94v5.66H9.35V9h3.42v1.56h.04a3.75 3.75 0 0 1 3.37-1.85c3.6 0 4.27 2.37 4.27 5.46v6.28zM5.34 7.43a2.06 2.06 0 1 1 0-4.12 2.06 2.06 0 0 1 0 4.12zm1.78 13.02H3.56V9h3.56v11.45zM22.22 0H1.78C.8 0 0 .77 0 1.73v20.54C0 23.23.8 24 1.77 24h20.45c.98 0 1.78-.77 1.78-1.73V1.73C24 .77 23.2 0 22.22 0z"
              />
            </svg>
            <span>
              Link to linkedin profile
            </span>
          </a>
          <a
            class="c6"
            color="#EDEDED"
            href="https://codepen.io/asjas"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              aria-labelledby="codepen-icon"
              viewBox="0 0 30 30"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24 8.18l-.02-.08-.02-.05c0-.03-.01-.06-.03-.08l-.02-.05-.03-.07-.03-.05-.04-.06-.05-.04-.06-.05-.05-.03-.06-.04-.04-.04-.01-.02L12.57.19c-.35-.23-.8-.23-1.14 0L.45 7.5l-.01.02-.05.03-.06.05-.03.04-.05.06-.04.04-.05.06c-.02.02-.03.03-.03.05l-.05.06-.02.06c-.02 0-.02.04-.03.07l-.01.05C0 8.12 0 8.15 0 8.18v7.5l.01.13.01.05.02.08.01.05.03.08.02.05.03.06.03.04c.02 0 .03.04.05.06l.03.04.04.04c0 .01 0 .03.03.03l.06.04.04.03v.01L11.4 23.8c.16.12.37.17.57.17s.39-.06.57-.18l10.99-7.28.01-.01.05-.04.06-.04.05-.04.05-.06.03-.04.04-.06.03-.05.03-.07.02-.05.03-.08.01-.05.03-.08v-7.5l-.01-.14-.02-.04h.05zm-11.99 6.28l-3.65-2.44 3.65-2.44 3.65 2.44-3.65 2.44zM10.98 7.8L6.5 10.78 2.9 8.36l8.09-5.39V7.8zm-6.33 4.23l-2.59 1.73V10.3l2.59 1.73zm1.85 1.25l4.48 3v4.81L2.89 15.7l3.62-2.42zm6.54 3l4.47-2.99 3.62 2.42-8.09 5.4v-4.83zm6.33-4.24l2.58-1.72v3.46l-2.58-1.73zm-1.86-1.24L13.04 7.8V2.97l8.09 5.39-3.61 2.42z"
              />
            </svg>
            <span>
              Link to codepen profile
            </span>
          </a>
        </footer>
      </nav>
    </section>
    <main
      class="c7"
    >
      <section
        class="c8"
      >
        <h1
          class="c9"
          data-testid="404-heading"
        >
          PAGE NOT FOUND
        </h1>
        <p
          class="c10"
        >
          You just hit a route that doesn't exist... the sadness.
        </p>
        <a
          class="c11"
          data-testid="404-link"
          href="/"
        >
          Go Home
        </a>
      </section>
    </main>
  </div>
</div>
`;
